File: app/src/main/java/com/example/myapplication/Adapter.java
package com.example.myapplication;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.ArrayList;
import java.util.List;
//Пример адаптера для ReciclerView
public class Adapter extends RecyclerView.Adapter<Adapter.MyViewHolder>{
    //наш массив с данными
    private ArrayList<MyDataModel> mData;

//инициализация классса
    public Adapter(Main3Activity main3Activity, ArrayList<MyDataModel> data) {
        this.mData =data;
    }

    @Override
    //устанавливаем внешний вид из layout xml файла
    public MyViewHolder onCreateViewHolder(ViewGroup viewGroup, int i) {
        return new MyViewHolder(LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.recycler_row, viewGroup, false));
    }

    @Override
    //указываем, что происходит при загрузке данных
    public void onBindViewHolder(MyViewHolder myViewHolder, int i) {
        MyDataModel data;
        data=mData.get(i);
        myViewHolder.R_ID.setText(data.getId());
        myViewHolder.R_Firstname.setText(data.getFirstname());
        myViewHolder.R_Lastname.setText(data.getLastname());
        myViewHolder.R_Group.setText(data.getGr());

    }

    @Override
    //указываем размер массива отображаемых данных
    public int getItemCount() {
        return mData.size();
    }
//назначаем ViewHolder
    public static class MyViewHolder extends RecyclerView.ViewHolder {

        TextView R_ID,R_Firstname,R_Lastname,R_Group;

        public MyViewHolder(View itemView) {
            super(itemView);

            R_ID = (TextView) itemView.findViewById(R.id.t_c1);
            R_Firstname = (TextView) itemView.findViewById(R.id.t_c2);
            R_Lastname = (TextView) itemView.findViewById(R.id.t_c3);
            R_Group = (TextView) itemView.findViewById(R.id.t_c4);
        }
    }

}

File: app/src/main/java/com/example/myapplication/Adapter.kt
package com.example.myapplication

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.myapplication.Adapter.MyViewHolder
import java.util.*

//Пример адаптера для ReciclerView
class Adapter(val mData: MutableList<MyDataModel>) : RecyclerView.Adapter<MyViewHolder>() {

    //устанавливаем внешний вид из layout xml файла
    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): MyViewHolder {
        return MyViewHolder(LayoutInflater.from(viewGroup.context).inflate(R.layout.recycler_row, viewGroup, false))
    }

    //указываем, что происходит при загрузке данных
    override fun onBindViewHolder(myViewHolder: MyViewHolder, i: Int) {
        val data: MyDataModel
        data = mData[i]
        myViewHolder.R_ID.text = data.id
        myViewHolder.R_Firstname.text = data.firstname
        myViewHolder.R_Lastname.text = data.lastname
        myViewHolder.R_Group.text = data.gr
    }

    //указываем размер массива отображаемых данных
    override fun getItemCount(): Int {
        return mData.size
    }

    //назначаем ViewHolder
    class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var R_ID: TextView
        var R_Firstname: TextView
        var R_Lastname: TextView
        var R_Group: TextView

        init {
            R_ID = itemView.findViewById<View>(R.id.t_c1) as TextView
            R_Firstname = itemView.findViewById<View>(R.id.t_c2) as TextView
            R_Lastname = itemView.findViewById<View>(R.id.t_c3) as TextView
            R_Group = itemView.findViewById<View>(R.id.t_c4) as TextView
        }
    }

}